lib/trie/database.go

```diff
@@ -287,3 +287,49 @@ func (t *Trie) writeDirty(db chaindb.Batch, curr node) error {
 	curr.setDirty(false)
 	return nil
 }
+
+// GetInsertedNodeHashes returns the hash of nodes that are inserted into state trie since last snapshot is called
+// Since inserted  nodesare newly created we need to compute their hash values.
+func (t *Trie) GetInsertedNodeHashes(curr node) ([]*common.Hash, error) {
+	var nodeHashes []*common.Hash
+	if curr == nil || !curr.isDirty() {
+		return nil, nil
+	}
+
+	enc, hash, err := curr.encodeAndHash()
+	if err != nil {
+		return nil, err
+	}
+
+	if curr == t.root {
+		h, err := common.Blake2bHash(enc) //nolint
```

Comment:
i think the `hash` returned by `encodeAndHash` should be the same as this unless `enc` is under 32 bytes, maybe change to check if `len(enc) < 32`?

Reply:
Added this condition `curr == t.root && len(enc) < 32`